### PR TITLE
Python: added parameter to control which context is used for the function call

### PR DESCRIPTION
**Pull request by eavanvalkenburg**
### Motivation and Context

Small improvement in the function calling setup, this allows you to control the context used during the function call, if you have functions that depend on static context (like text_memory_skill with the collection name) then you need the existing context with the additional parameters, this is now possible. The default is that a new context is created and used, same as before.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/samples/kernel-syntax-examples/openai_function_calling.py [hunk 1]
```diff
@@ -34,13 +32,6 @@
     ),
 )
 
-skills_directory = os.path.join(__file__, "../../../../samples/skills")
```
Comment (moonbox3) [inline python/samples/kernel-syntax-examples/openai_function_calling.py:37]: This is an intentional removal of this skill configuration?

File: python/semantic_kernel/connectors/ai/open_ai/utils.py [hunk 2]
```diff
@@ -167,7 +174,17 @@ async def chat_completion_with_function_call(
     # if there is no function_call or if the content is not a FunctionCall object, return the context
     if function_call is None or not isinstance(function_call, FunctionCall):
         return context
-    result = await execute_function_call(kernel, function_call, log=log)
+    if function_call_with_new_context:
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/open_ai/utils.py:177]: Are there any test cases that need to be added or updated related to this change?

File: python/samples/kernel-syntax-examples/openai_function_calling.py (hunk 1 above)
Comment (eavanvalkenburg) [reply python/samples/kernel-syntax-examples/openai_function_calling.py]: yes, more of an oversight that it was in there in the first place, it is not usefull for the sample, so should be deleted to make sure the sample is as easy as possible. Could be a seperate PR...

File: python/semantic_kernel/connectors/ai/open_ai/utils.py (hunk 2 above)
Comment (eavanvalkenburg) [reply python/semantic_kernel/connectors/ai/open_ai/utils.py:177]: This utility func is currently not tested, I will have a look.

Comment (moonbox3) [reply python/semantic_kernel/connectors/ai/open_ai/utils.py:177]: We can always create a separate item to track adding tests.

**Review by moonbox3 — APPROVED** [review]
The rest looks good to me.

File: python/semantic_kernel/connectors/ai/open_ai/utils.py (hunk 2 above)
Comment (eavanvalkenburg) [reply python/semantic_kernel/connectors/ai/open_ai/utils.py:177]: sounds good, will start that asap

**Review by alexchaomander — APPROVED** [review]
LGTM!

**Pull request closed without merging**